src/App.vue

```diff
@@ -1,14 +1,104 @@
 <template>
-  <Base/>
+  <v-app light>
+    <Navbar :menu-items="menuItems" />
+    <MobileNavbar :menu-items="menuItems" />
+    <vue-page-transition style="height:100%" >
+    <component :is="layout">
+      <vue-page-transition style="height:100%" >
+        <router-view/>
+      </vue-page-transition>
+    </component>
+    </vue-page-transition>
+  </v-app>
 </template>
 
 <script>
-import Base from './views/Base'
+import Default from '@/components/layouts/Default.vue'
+import Large from '@/components/layouts/Large.vue'
+import None from '@/components/layouts/None.vue'
```

Comment:
Nie rozumiem trochę tej idei, możesz ją przedstawić?

Reply:
Co to znaczy None Layout np.?

Reply:
Przy tworzeniu viewsów napotkałem problem, ponieważ w komponencie bazowym dało się zdefiniować tylko jeden layout. View home i kreatora quizów korzystały z innego layoutu, a reszta z innego. Jednym z rozwiązań byłoby wprowadzanie layoutu na początku każdego viewsa, ale wiązało by się to z duplikacją kodu i dużą ilością zmian, gdy zechciałbym zmienić layout. To rozwiązanie pozwala definiować layout w routerze za pomocą metainformacji związanych z routem. Gdy uzytkownik wchodzi na danego routa odczytywana jest ta informacja i na jej podstawie dynamicznie ładowany jest wybrany layout zdefiniowany w tych plikach. Rozwiązuje to powyższy problem.

Reply:
W Nuxt też jest coś takiego tylko definiuje się to przy exporcie viewsa.

Reply:
Ekstra tyllko nienazywaj layoutu none.vue